common/src/seg_tree.c

```diff
@@ -217,6 +217,40 @@ int seg_tree_add(struct seg_tree* seg_tree, unsigned long start,
     return 0;
 }
 
+/* search tree for entry that overlaps with given start/end
+ * offsets, return first overlapping entry if found, NULL otherwise,
+ * assumes caller has lock on tree */
```

Comment:
Is there a reason why you don't want to do the locking inside of this function?  All other seg_tree functions do the locking for you, unless there's an explicit reason for them not to.  Also, please capitalize sentences.

Reply:
The motivation here was thinking about cases where one would be looking up a large set of extents, where this would be called in a loop where locking outside of the loop would help with performance:
```
lock tree
foreach extent in many extents
  seg_tree_find(extent)
unlock tree
```
I agree that we should make things consistent, though, so let me check on this again.

Reply:
I don't think this is going to be an issue.  The lock acquisition is going to pale in comparison to the number of cycles needed for the tree traversal.   We can always add a `seg_tree_find_nolock()` if performance testing shows that it is a bottleneck.

Reply:
Ok, checking the code, actually there is a second reason I did this.  The actual usage looks like this:
```
lock tree
seg_tree_find
iterate over segments to compute something
seg_tree_find
iterate over segments again
unlock tree
```
The first loop iterates over the extents to figure something out, and if true, iterates over them again to do something with them.  To ensure consistency, it holds a read lock through all of this, so the lock happens on the outside of the find/iterate loops.

Reply:
See the code in service_local_reqs() in unifyfs-sysio.c.  So I think it has to be done that way.

Reply:
Ok, that's a legit use case.  Can you have this patch add both variants then:
```
seg_tree_find()
seg_tree_find_nolock()  (or whatever you want to name this)
```

Reply:
I'll compromise with you.  I renamed the existing function to set_tree_find_nolock().  I'll defer adding seg_tree_find() to a later PR since we're not using it yet.